rust-runtime/aws-smithy-runtime/src/client/orchestrator/auth.rs

```diff
@@ -124,6 +125,11 @@ fn extract_endpoint_auth_scheme_config(
     endpoint: &Endpoint,
     scheme_id: AuthSchemeId,
 ) -> Result<AuthSchemeEndpointConfig<'_>, AuthOrchestrationError> {
+    // TODO(P96049742): Endpoint config doesn't currently have a concept of optional auth or "no auth", so
+    // we are short-circuiting lookup of endpoint auth scheme config if that is the selected scheme.
+    if scheme_id == NO_AUTH_SCHEME_ID {
```

Comment:
I might consider renaming this—I'm not sure if `NO_AUTH_SCHEME_ID` means "empty" or "not authenticated". This would probably be clearer as

```AuthSchemeId::unauthenticated()```

Reply:
It's referring to the special auth scheme named "no auth", but I decided to leave out the second "auth" in the name: `NO_AUTH_AUTH_SCHEME_ID`.